packages/sanity/src/structure/panes/document/useDocumentTitle.ts

```diff
@@ -50,6 +55,15 @@ export function useDocumentTitle(): UseDocumentTitle {
     }
   }
 
+  // in cases where the displayed document has been slated to be unpublished
+  // then the displayed title should be the name of the document
+  if (isGoingToUnpublish(displayed as SanityDocument)) {
```

Comment:
My initial approach was to use the condition `if ('_system' in displayed && (displayed as Partial<SanityDocument>)._system?.delete))` but this felt better since it's a condition we use in other places to check the same use case. Let me know if you have a strong preference of one over the other

```suggestion
  if ('_system' in displayed && (displayed as SanityDocument)._system?.delete) {
```

Reply:
- I don't think it makes sense (when the document is slated for unpublishing) that we show the title of the "published", shouldn't it still show the information for the version that is being displayed?
- It's not _always_ the published doc (for example in situations where the release will have the document published before in the perspective stack, the way we handle it in other places is by showing what the document will look like with the stack before this one that you are trying to unpublish). We would need to go through the stack and see what document it actually is (and that would not be available through the editState here).

This feels like it touches on a larger problem / question I guess that is not necessarily related to the titles but to the document displayed:
When a document is slated for unpublishing, the displayed document is _not_ the document that will come next on the stack (on the example video, Kayla's version). It will be the version of the release itself, for example: if I created a version of a document and edited it, and _then_ click to unpublish -> the displayed version will still be that version _not_ whatever comes next in perspective stack. We could make this change, but I'm not sure if this won't add confusion :/ 

Hopefully the video and notes explain what I mean:

https://github.com/user-attachments/assets/249125e4-7ef6-4584-b2e8-28aa5ab13689



As far as I know a document will _always_ have name. The title however, is optional, but I don't think that's necessarily a bad thing since we do have fall backs for when the title is undefined where it shows the name "untitled" but those should still work as expected

Reply:
Or maybe it does make sense to link the whole displayed document to the previous stack version? The more I think about this the less sure i feel 😂 but in which case then the link from the releases detail should also link to it instead of the release document version with the banner. But again, I'm not sure if that just won't add to the confusion tbh

Or maybe it's about updating also the text that is on the banner